router/src/server.rs

```diff
@@ -111,11 +111,21 @@ example = json ! ({"error": "Incomplete generation"})),
 async fn completions_v1(
     default_return_full_text: Extension<bool>,
     infer: Extension<Infer>,
+    headers: HeaderMap,
     req: Json<CompletionRequest>,
 ) -> Result<Response, (StatusCode, Json<ErrorResponse>)> {
     let req = req.0;
     let mut gen_req = CompatGenerateRequest::from(req);
 
+    // If a bearer token was present in this request, pass the associated token along.
+    let _ = headers.get("authorization").map_or((), |x| {
```

Comment:
We should do this for the generate and generate_stream endpoints as well (if the user did not pass a token explicitly).

Reply:
Done.